### PR TITLE
Docker multi-arch images push for linux/amd64, linux/arm64

### PR DESCRIPTION
**Pull request by salman2013**
In this PR i updated github action to push multi-architecture(amd-64, arm-64) docker images on the docker hub. Now the M1/M2 apple machines would pull the arm-64 architecture docker images via devstack.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @salman2013!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.




:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by editing the PR title.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

**Comment by UsamaSadiq** [comment]
Hi @mphilbrick211 
Salman is a new member of the @openedx/arbi-bom team (2U) and his cla creation process is currently in-progress. Jeremy has created a support issue for it so it'll get resolved pretty soon.

**Comment by mphilbrick211** [comment]
> Hi @mphilbrick211 Salman is a new member of the @openedx/arbi-bom team (2U) and his cla creation process is currently in-progress. Jeremy has created a support issue for it so it'll get resolved pretty soon.

Thank you for the update, @UsamaSadiq!

File: .github/workflows/push-docker-image.yml [hunk 1]
```diff
@@ -29,22 +29,30 @@ jobs:
...
+        uses: docker/setup-qemu-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+           username: ${{ secrets.DOCKERHUB_USERNAME }}
+           password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
       - name: Build and push base Docker image
         uses: docker/build-push-action@v1
```
Comment (jmbowman) [inline .github/workflows/push-docker-image.yml:45]: Should we upgrade the action version here also for consistency?

File: .github/workflows/push-docker-image.yml [hunk 2]
```diff
@@ -29,22 +29,30 @@ jobs:
...
         with:
           push: ${{ github.event_name != 'pull_request' }}
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_PASSWORD }}
           target: base
-          repository: openedx/credentials
```
Comment (jmbowman) [inline .github/workflows/push-docker-image.yml:39]: Where is this configured now?  I'm not seeing it in the Dockerfile.

File: .github/workflows/push-docker-image.yml (hunk 1 above)
Comment (salman2013) [reply .github/workflows/push-docker-image.yml:45]: yes we should update as per https://github.com/docker/build-push-action

File: .github/workflows/push-docker-image.yml (hunk 2 above)
Comment (salman2013) [reply .github/workflows/push-docker-image.yml:39]: could you please elaborate on this comment?

File: .github/workflows/push-docker-image.yml (hunk 1 above)
Comment (jmbowman) [reply .github/workflows/push-docker-image.yml:45]: Right, I was just wondering why it was updated in this PR for the dev image but not the base image.

File: .github/workflows/push-docker-image.yml (hunk 2 above)
Comment (salman2013) [reply .github/workflows/push-docker-image.yml:39]: @jmbowman Previously i removed the repository which i followed as per the following documentation but now i am realizing it should be added so i added it again. The reason is that the documentation is mentioning the `context .` but in our case we have to build one for the base and one for dev.  

https://docs.docker.com/build/ci/github-actions/multi-platform/

File: .github/workflows/push-docker-image.yml (hunk 1 above)
Comment (salman2013) [reply .github/workflows/push-docker-image.yml]: I have updated it.

**Review by jmbowman — APPROVED** [review]

File: .github/workflows/push-docker-image.yml (hunk 2 above)
Comment (jmbowman) [reply .github/workflows/push-docker-image.yml:39]: Just noting that you said you added it back, but it still seems to be removed in the PR in both the base and dev images.

Comment (salman2013) [reply .github/workflows/push-docker-image.yml:39]: Yes, i reviewed it again, there was some issue with my commit, i fixed it now. Thanks for notifying me of this.

**Pull request merged**